public/api/updateAssignmentSettingsByKey.php

```diff
@@ -0,0 +1,138 @@
+  ";
+  $result = $conn->query($sql);
+  if ($result->num_rows > 0) {
+    $row = $result->fetch_assoc();
+    $courseId = $row['courseId'];
+  }else{
+    throw new Exception("Internal Error: missing courseId.");
+
+  }
+}
+
+
+$permissions = permissionsAndSettingsForOneCourseFunction($conn,$userId,$courseId);
+if ($permissions["canModifyActivitySettings"] != '1'){
+  throw new Exception("You need permission to edit content.");
+}
+
+$settingKeys = array(
+  "individualize","showSolution","timeLimit","showFeedback","showHints","showCorrectness",
+  "showCreditAchievedMenu","paginate","showFinishButton","autoSubmit","canViewAfterCompleted",
+  "assignedDate","dueDate","pinnedUntilDate","pinnedAfterDate","proctorMakesAvailable",
+  "totalPointsOrPercent","showSolutionInGradebook","gradeCategory","attemptAggregation",
+  "numberOfAttemptsAllowed"
+);
+
+$providedValues = [];
+
+foreach ($settingKeys as $key) {
+  if(array_key_exists($key, $_POST)) {
+    $providedValues[$key] = mysqli_real_escape_string($conn,$_POST[$key]);
+  }
+}
+unset($key, $value);
+
+//protect against invalid empty string values
+if (array_key_exists("pinnedAfterDate", $providedValues) && $providedValues["pinnedAfterDate"] == ''){$providedValues["pinnedAfterDate"] = 'NULL';}
+if (array_key_exists("pinnedUntilDate", $providedValues) && $providedValues["pinnedUntilDate"] == ''){$providedValues["pinnedUntilDate"] = 'NULL';}
+if (array_key_exists("timeLimit", $providedValues) && $providedValues["timeLimit"] == ''){$providedValues["timeLimit"] = 'NULL';}
+if (array_key_exists("dueDate", $providedValues) && $providedValues["dueDate"] == ''){$providedValues["dueDate"] = 'NULL';}
+if (array_key_exists("assignedDate", $providedValues) && $providedValues["assignedDate"] == ''){$providedValues["assignedDate"] = 'NULL';}
+if (array_key_exists("totalPointsOrPercent", $providedValues) && $providedValues["totalPointsOrPercent"] == '') { $providedValues["totalPointsOrPercent"] = 'NULL';}
+if (array_key_exists("gradeCategory", $providedValues) && $providedValues["gradeCategory"] == '') { $providedValues["gradeCategory"] = 'NULL';}
+if (array_key_exists("numberOfAttemptsAllowed", $providedValues) && $providedValues["numberOfAttemptsAllowed"] == ''){$providedValues["numberOfAttemptsAllowed"] = 'NULL';}
+
+//protect against invalid boolean values
+$boolKeys = array(
+  "individualize","showSolution","showFeedback","showHints","showCorrectness",
+  "showCreditAchievedMenu","paginate","showFinishButton","autoSubmit","canViewAfterCompleted",
+  "proctorMakesAvailable","showSolutionInGradebook"
+);
+
+foreach ($boolKeys as $key) {
+  if (array_key_exists($key, $providedValues)) {
+    if ($providedValues[$key]) {$providedValues[$key] = '1';} else {$providedValues[$key] = '0';}
+  }
+}
+
+function array_map_assoc( $callback , $array ){
+  $r = array();
+  foreach ($array as $key=>$value)
+    $r[$key] = $callback($key,$value);
+  return $r;
+}
+;
+
+$updates = implode(
+  ",", 
+  array_map_assoc(
```

Comment:
would be nicer to share this code with similar block in updateContentFlags.php, there are several things here that are copy-pasted from there and updateAssignmentSettings.php